pkg/platform/runtime/buildexpression/buildexpression.go

```diff
@@ -171,6 +171,37 @@ func New(data []byte) (*BuildExpression, error) {
 	return expr, nil
 }
 
+// NewFromPlatformIDAndRequirements creates a minimal buildexpression from the given platform UUID
+// and requirement list.
+func NewFromPlatformIDAndRequirements(platformID string, requirements []model.Requirement) (*BuildExpression, error) {
+	// At this time, there is no way to ask the Platform for an empty buildexpression, so build one
+	// manually and then add a timestamp, platform, and language requirement to it.
+	expr, err := New([]byte(fmt.Sprintf(`
+		{
+			"let": {
+				"runtime": {
+					"solve_legacy": {
+						"at_time": "%s",
+						"build_flags": [],
+						"camel_flags": [],
+						"platforms": ["%s"],
+						"requirements": [],
+						"solver_version": null
+					}
+				},
+				"in": "$runtime"
+			}
+		}
+	`, time.Now().Format(time.RFC3339), platformID)))
```

Comment:
We need to source the timestamp from the inventory API when we make updates like this. If we push it too far there could be unforeseen issues. If you look at other calls to `StageCommit` we are using this function to update the timestamp: https://github.com/ActiveState/cli/blob/master/pkg/platform/model/inventory.go#L404